Solaris specific customizations to session startup script
Not suitable for upstream

Oracle Bug 15700113 - SUNBT7023672 The vncserver is unable to start a typical
       gnome Solaris 11 desktop

Oracle Bug 25022410 - vnc changes for gnome3

--- a/xinitrc.in	2025-09-06 11:36:14.008132935 -0700
+++ b/xinitrc.in	2025-09-06 11:37:57.216755467 -0700
@@ -40,6 +40,13 @@
     "$xmodmap" "$usermodmap"
 fi
 
+# If GNOME desktop is available, run it
+# In GNOME 3, user unable to enter the password without ck-launch-session
+#
+if [ -x /etc/gdm/Xsession ] && [ -x /usr/bin/dtstart ] ; then
+  exec /usr/bin/ck-launch-session /usr/bin/dtstart jds
+fi
+
 # start some nice programs
 
 if [ -d "$xinitdir"/xinitrc.d ] ; then
